martin/src/source.rs

```diff
@@ -28,7 +31,91 @@ impl Display for Xyz {
 
 pub type Tile = Vec<u8>;
 pub type UrlQuery = HashMap<String, String>;
-pub type Sources = HashMap<String, Box<dyn Source>>;
+
+#[derive(Default, Clone)]
+pub struct Sources(HashMap<String, Box<dyn Source>>);
+
+impl Sources {
+    pub fn insert(&mut self, id: String, source: Box<dyn Source>) {
+        self.0.insert(id, source);
```

Comment:
nit: It might be worth giving the `HashMap` in `Sources` a name, so that the accesses are clearer

Reply:
this is a fairly common pattern - esp when there is only one value wrapped. I would rather not change it to a named struct